### ISSUE TITLE
To avoid stressing the Exact Online docs an html cache should be introduced.

### ISSUE THREAD

**Issue by DannyvdSluijs** [DannyvdSluijs/exact-online-meta-data-tool#187]
To avoid stressing the Exact Online docs an html cache should be introduced. This would also speed up a secondary run when you're working on local changes which don't require refetching of the HTML. An option should be included to skip caching

**Comment by DannyvdSluijs** [comment]
Something like 
```php
private function fetchHtmlFromUrl(string $url): string
    {
        [,$basename] = explode('=', $url);
        $filename = sys_get_temp_dir() . '/exact-online-meta-data-tool-' . strtolower($basename) . '.html';

        if (!file_exists($filename)) {
            $html = file_get_contents($url);
            file_put_contents($filename, $html);
        } else {
            $html = file_get_contents($filename);
        }

        $this->visitedPages->add($url);

        if ($html === false) {
            throw new \RuntimeException('Unable to fetch html from ' . $url);
        }

        return $html;
    }
```

Can already help.

**Comment by DannyvdSluijs** [comment]
Solved in #194
